common/screenshotter.go

```diff
@@ -61,11 +61,12 @@ func (f *ImageFormat) UnmarshalJSON(b []byte) error {
 }
 
 type screenshotter struct {
-	ctx context.Context
+	ctx       context.Context
+	persister storage.FilePersister
```

Comment:
Great work on the PR! Just a thought – how about we kick off by implementing our local and remote persisters before declaring the `persister` interface and adjusting the `screenshotter`? This way, we can avoid redoing stuff and streamline our PR process.

If that plan doesn't fit, what if we introduce the interface right here instead of using `FilePersister`? Since `screenshotter` will be switching between different implementations, placing the interface in the `common` package seems like a smart move. This would also make swapping implementations in our mapping layer and tests easier.

What do you think?

Reply:
I've made the change so that it is a stand alone type. The interface will come later when the remotePersister is finalised.

Reply:
👍